src/libponyc/type/subtype.c

```diff
@@ -689,24 +693,61 @@ static bool is_tuple_sub_single(ast_t* sub, ast_t* super, check_cap_t check_cap,
   if(errorf != NULL)
   {
     ast_error_frame(errorf, sub,
-      "%s is not a subytpe of %s: the subtype is a tuple",
+      "%s is not a subtype of %s: the supertype is a tuple",
       ast_print_type(sub), ast_print_type(super));
   }
 
   return false;
 }
 
-static bool is_single_sub_tuple(ast_t* sub, ast_t* super, check_cap_t check_cap,
-  errorframe_t* errorf, pass_opt_t* opt)
+static bool is_tuple_sub_nominal(ast_t* sub, ast_t* super,
+  check_cap_t check_cap, errorframe_t* errorf, pass_opt_t* opt)
 {
-  (void)check_cap;
-  (void)opt;
+  ast_t* super_def = (ast_t*)ast_data(super);
+
+  if(ast_id(super_def) == TK_INTERFACE)
+  {
+    ast_t* super_members = ast_childidx(super_def, 4);
+    pony_assert(ast_id(super_members) == TK_MEMBERS);
+
+    if(ast_childcount(super_members) == 0)
+    {
+      // This is an empty interface, so we let the tuple be a subtype if the
+      // caps of the elements of the tuple are subcaps of the supertype cap.
+      for(ast_t* child = ast_child(sub);
+        child != NULL;
+        child = ast_sibling(child))
+      {
+        if(!is_sub_cap_and_eph(child, super, check_cap, errorf, opt))
```

Comment:
Instead of `is_sub_cap_and_eph`, I would make this `is_x_sub_x(child, super, ...)`